osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/Event.java

```diff
@@ -10,32 +10,53 @@
 
 import java.io.Serializable;
 
+import lombok.Getter;
 import org.joda.time.DateTime;
 
+@Getter
 public class Event implements Serializable {
 
-    /**
-     * Serial Version UID.
-     */
     private static final long serialVersionUID = 4482313912422705642L;
     private final DateTime timestamp;
     private final Integer eventCode;
-    private final Integer eventCounter;
     private final EventLogCategory eventLogCategory;
+    private final Integer eventCounter;
+    private final DateTime startTime;
+    private final Long duration;
 
     public Event(final DateTime timestamp, final Integer eventCode, final Integer eventCounter,
             final EventLogCategory eventLogCategory) {
         this.timestamp = timestamp;
         this.eventCode = eventCode;
         this.eventCounter = eventCounter;
         this.eventLogCategory = eventLogCategory;
+        this.startTime = null;
+        this.duration = null;
+    }
+
+    public Event(final DateTime endTime, final Integer eventCode, final EventLogCategory eventLogCategory,
+            final DateTime startTime, final Long duration) {
+        this.timestamp = endTime;
+        this.eventCode = eventCode;
+        this.eventCounter = null;
+        this.eventLogCategory = eventLogCategory;
+        this.startTime = startTime;
+        this.duration = duration;
     }
 
     @Override
     public String toString() {
-        return String.format("Event[time=%s, code=%s, category=%s, type=%s%s]", this.timestamp, this.eventCode,
-                this.eventLogCategory.name(), this.eventCode == null ? null : EventType.getByEventCode(this.eventCode),
-                this.eventCounter == null ? "" : ", counter=" + this.eventCounter);
+        if (eventLogCategory == EventLogCategory.POWER_FAILURE_EVENT_LOG) {
```

Comment:
this is very ugly. as stated earlier you can just make the power_failure items optional,

Reply:
I totally agree, but passing null values to a constructor isn't that nice either.